### PR TITLE
Add JSXText check into isValidLocationToAddComment

### LINKED ISSUE #27552: "Ignore this error message" code fix in JSX results in rendering `// @ts-ignore`
**TypeScript Version:**  3.2.0-dev.20181004

**Search Terms:**

disableJsDiagnostics
JSX 
Code fix
Ignore this error message
Add '@ts-ignore' to all error messages

**Code**

```jsx
// MyComponent.jsx
// @ts-check
import React from "react";

class MyComponent extends React.Component {
  render() {
    return (
      <div>
        // @ts-ignore
        {doesNotExist}
      </div>
    );
  }
}

export default MyComponent;
```

Running the `Ignore this error message` or `Add '@ts-ignore' to all error messages` code fix inserts a `// @ts-ignore` which satisfies the compiler.

But,

```
<div>
  // @ts-ignore
  {doesNotExist}
</div>
```

will actually render `// @ts-ignore`.

**Expected behavior:**

Looks like `{/* @ts-ignore */}` or `{/* // @ts-ignore */}` are not recognized as valid ignore comments.

So, the best I could come up with is

```
<div>
  {/* 
  // @ts-ignore */}
  {doesNotExist}
</div>
```

**Actual behavior:**

```jsx
// MyComponent.jsx
// @ts-check
import React from 'react';

class MyComponent extends React.Component {
  render() {
    return (
      <div>
        // @ts-ignore
        {doesNotExist}
      </div>
    );
  }
}

export default MyComponent;
```

where `// @ts-ignore` mistakenly gets rendered.

**Related Issues:** 

https://github.com/Microsoft/TypeScript/issues/25240

### PR DESCRIPTION
**Pull request by weswigham**
Fixes #27552

### CODE REVIEW

File: src/services/utilities.ts [hunk 1]
```diff
@@ -906,6 +906,20 @@ namespace ts {
         return isTemplateLiteralKind(token.kind) && position > token.getStart(sourceFile);
     }
 
+    export function isInJSXText(sourceFile: SourceFile, position: number) {
+        const token = getTokenAtPosition(sourceFile, position);
+        if (isJsxText(token)) {
+            return position >= token.getStart(sourceFile);
```
Comment (ajafff) [inline src/services/utilities.ts]: You probably want to use `token.pos` because `getStart` will handle parts of the JsxText as trivia.

Comment (ajafff) [reply src/services/utilities.ts]: on second thought: is this even necessary? JsxText cannot have leading trivia, therefore you can always `return true;` here

File: tests/cases/fourslash/jsxTsIgnoreOnJSXExpressionsAndChildren.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+goTo.file(3);
+verify.codeFix({
+    index: 0,
+    description: "Ignore this error message",
+    newFileContent: `class MyComponent3 extends React.Component {
+  render() {
+    return (
+      <div>
+        <
+// @ts-ignore
```
Comment (oshalygin) [inline tests/cases/fourslash/jsxTsIgnoreOnJSXExpressionsAndChildren.ts]: vscode completely loses all sense of intellisense when this happens.  Seems fine though on the rendering side.  Is this a case where we should file an issue against the VSCode repo?

@weswigham any input <3 ?

Comment (weswigham) [reply tests/cases/fourslash/jsxTsIgnoreOnJSXExpressionsAndChildren.ts]: If it still happens reliably and you can provide some complete repro steps, a new issue would be pertinent.

**Pull request merged**